Examples/test-suite/c/operator_overload_runme.c

```diff
@@ -0,0 +1,26 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "operator_overload/operator_overload_wrap.h"
+
+#define assert(x,msg) if (!x) { printf("%d: %s\n", x, msg); exit(1); }
```

Comment:
This evaluates `x` twice which is not good if it has side-effects.

It also lacks parentheses so this assert doesn't fail (because the condition evaluates like `((!1) == 2)`):

```
assert(1==2, "oops");
```

It also doesn't swallow a following `;` which can give surprising control flow in some cases (and confusing compilation errors in others) if `assert` is used in an unbraced block.

Why don't we use `<assert.h>` here rather than reinventing a rather buggy alternative version?

The standard `assert()` doesn't directly support a message, but one can just put a comment before it, which should be fine here as the testcases are not huge programs.

Or there are common hacks like the one noted at https://en.cppreference.com/w/cpp/error/assert:

```
assert(("There are five lights", 2 + 2 == 5));
```

You'd probably need to cast the string literal to `void` to avoid compiler warnings, but something like this works:

```
#include <assert.h>

#define assert2(x, msg) assert(((void)msg, x))
```

E.g. with GCC that gives me a warning free compilation with `-Wall -W -O2` and runtime output of:

```
$ ./a.out 
a.out: badassert.c:9: main: Assertion `((void)"boo", 1==2)' failed.
Aborted
```

With clang it's very similar:

```
$ ./a.out 
a.out: badassert.c:9: int main(): Assertion `((void)"boo", 1==2)' failed.
Aborted
```

Reply:
I see that most of the other testcases just use the standard assert, so probably just doing that here too makes most sense.

Reply:
I have no idea why does this test, and it only, use a custom `assert` macro, but I'll get rid of it.